modules/swagger-codegen/src/main/resources/typescript-angular/configuration.mustache

```diff
@@ -23,4 +23,53 @@ export class Configuration {
         this.basePath = configurationParameters.basePath;
         this.withCredentials = configurationParameters.withCredentials;
     }
+
+    /**
+     * Select the correct content-type to use for a request.
+     * Uses {@link Configuration#isJsonMime} to determine the correct content-type.
+     * If no content type is found return the first found type if the contentTypes is not empty
+     * @param {string[]} contentTypes - the array of content types that are available for selection
+     * @returns {string} the selected content-type or <code>undefined</code> if no selection could be made.
+     */
+    public selectHeaderContentType (contentTypes: string[]): string {
+        if (contentTypes.length == 0)
+            return undefined;
+
+        let type = contentTypes.find(x => this.isJsonMime(x));
+        if (type === undefined) 
+            return contentTypes[0];
+        return type;
+    }
+
+    /**
+     * Select the correct accept content-type to use for a request.
+     * Uses {@link Configuration#isJsonMime} to determine the correct accept content-type.
+     * If no content type is found return the first found type if the contentTypes is not empty
+     * @param {string[]} accepts - the array of content types that are available for selection.
+     * @returns {string} the selected content-type or <code>undefined</code> if no selection could be made.
+     */
+    public selectHeaderAccept(accepts: string[]): string  {
+        if (accepts.length == 0)
+            return undefined;
+
+        let type = accepts.find(x => this.isJsonMime(x));
+        if (type === undefined) 
+            return accepts[0];
+        return type;
+    }
+
+    /**
+     * Check if the given MIME is a JSON MIME.
+     * JSON MIME examples:
+     *   application/json
+     *   application/json; charset=UTF8
+     *   APPLICATION/JSON
+     *   application/vnd.company+json
+     * @param {string} mime - MIME (Multipurpose Internet Mail Extensions)
+     * @return {boolean} True if the given MIME is JSON, false otherwise.
+     */
+    public isJsonMime(mime: string): boolean {
+        let jsonMimeRegEx = new RegExp("^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$");
```

Comment:
how about also supporting json patch mime types?

the latest version of the `isJsonMime()` method before it was removed can be found here:
https://github.com/swagger-api/swagger-codegen/pull/6574/files#diff-3ab1141e07abf8762d6e33d15c9d1f0eL63

```
    public isJsonMime(mime: string): boolean {		
        const jsonMime: RegExp = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');		
        return mime != null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');		
    }
```